crates/jstz_kernel/src/inbox.rs

```diff
@@ -145,3 +160,86 @@ pub fn read_message(
         },
     }
 }
+
+#[cfg(test)]
+mod test {
+    use jstz_mock::mock::{JstzMockHost, MockNativeDeposit};
+    use jstz_proto::operation::external;
+    use tezos_crypto_rs::hash::{ContractKt1Hash, HashTrait};
+    use tezos_smart_rollup::types::SmartRollupAddress;
+
+    use super::{read_message, InternalMessage, Message};
+
+    #[test]
+    fn read_message_ignored_on_different_smart_rollup_address() {
+        let mut host = JstzMockHost::new(true);
+        let alternative_smart_rollup_address =
+            SmartRollupAddress::from_b58check("sr1Ghq66tYK9y3r8CC1Tf8i8m5nxh8nTvZEf")
+                .unwrap();
+        let deposit = MockNativeDeposit {
+            smart_rollup: Some(alternative_smart_rollup_address),
+            ..MockNativeDeposit::default()
+        };
+        host.add_deposit_message(&deposit);
+        let ticketer = host.get_ticketer();
+        let result = read_message(host.rt(), ticketer);
+        assert_eq!(result, None)
+    }
+
+    #[test]
+    fn read_message_native_deposit_succeeds() {
+        let mut host = JstzMockHost::new(true);
+        let deposit = MockNativeDeposit::default();
+        let ticketer = host.get_ticketer();
+        host.add_deposit_message(&deposit);
+        match read_message(host.rt(), ticketer) {
+            Some(Message::Internal(InternalMessage::Deposit(external::Deposit {
+                amount,
+                reciever,
+            }))) => {
+                assert_eq!(amount, 100);
+                assert_eq!(reciever.to_base58(), deposit.receiver.to_b58check())
+            }
+            _ => panic!("Expected deposit message"),
+        };
```

Comment:
nit: Instead of matching you could do:
```rust
let Message::Internal(InternalMessage::Deposit(external::Deposit { ... })) = read_message(host.rt(), ticketer).expect("Expected deposit message");
```

Reply:
This is not possible because `Message` is an enum. I have changed it to `if let` instead, and handled both the empty option and else cases.